lib/dao.js

```diff
@@ -427,19 +495,65 @@ DataAccessObject.exists = function exists(id, cb) {
  * ```
  *
  * @param {*} id Primary key value
+ * @param {Object} [options] Options
  * @param {Function} cb Callback called with (err, instance)
  */
-DataAccessObject.findById = function find(id, cb) {
+DataAccessObject.findById = function find(id, options, cb) {
   if (stillConnecting(this.getDataSource(), this, arguments)) return;
-  this.findOne(byIdQuery(this, id), cb);
+
+  assert(arguments.length >= 1, 'The id argument is required');
+  if (arguments.length === 2) {
+    if (typeof options === 'function') {
+      // findById(id, cb)
+      cb = options;
+      options = {};
+    }
+  }
+  cb = cb || noCallback;
+  options = options || {};
+
+  assert(typeof options === 'object', 'The options argument must be an object');
+  assert(typeof cb === 'function', 'The cb argument must be a function');
+
+  if (id == null || id === '') {
+    return process.nextTick(function() {
+      cb(new Error('Model::findById requires the id argument'));
+    });
+  }
+  this.findOne(byIdQuery(this, id), options, cb);
 };
 
-DataAccessObject.findByIds = function(ids, cond, cb) {
-  if (typeof cond === 'function') {
-    cb = cond;
-    cond = {};
+/**
+ * Find model instances by ids
+ * @param {Array} ids An array of ids
+ * @param {Object} query Query filter
+ * @param {Object} [options] Options
+ * @param {Function} cb Callback called with (err, instance)
+ */
+DataAccessObject.findByIds = function(ids, query, options, cb) {
```

Comment:
Is `query` a full query, or is it just a `where` part? The name `cond` makes me to believe it's only the `where` part. (I did not check the implementation to verify.)